pkg/genericapiserver/options/server_run_options.go

```diff
@@ -157,6 +148,10 @@ func (o *ServerRunOptions) WithEtcdOptions() *ServerRunOptions {
 	o.Etcd = NewDefaultEtcdOptions()
 	return o
 }
+func (o *ServerRunOptions) WithSecureServingOptions() *ServerRunOptions {
```

Comment:
I expected a call to this in NewServerRunOptions. Alternatively, rename this to `WithSecureServing**Defaults**`.